Add Development only flag-notifier

mreishus opened the pull request:
#### Changes proposed in this Pull Request

* Add a development-only flag notifier
  * As we work on more projects with flags, makes it easier to devs to tell if a flag is on or not
  * Doesn't scale to 10+ flags but can be refactored if we need it
  * Doesn't affect customers

#### Testing instructions

* Wait for #47523 to be merged
* Visit calypso with and without "?flags=anchor-fm" in the url
* Look in the masterbar

![2020-11-17_12-21](https://user-images.githubusercontent.com/937354/99430811-7028de80-28cf-11eb-8d54-405828e63363.png)
![2020-11-17_12-21_1](https://user-images.githubusercontent.com/937354/99430820-728b3880-28cf-11eb-8c17-512bb69f0360.png)

mreishus committed:
Add Development only flag-notifier

kwight commented:
Hm, doesn't this serve the same purpose as the existing environment badge? Knowing that `anchor-fm` is only activated on Horizon (or `wp-calypso`, or whatever), we can quickly see if anchor.fm is active or not.

<img width="296" alt="Screen Shot 2020-11-17 at 12 01 56 PM" src="https://user-images.githubusercontent.com/349751/99441649-06a7d080-28cd-11eb-8f9e-58d256d9fcd2.png">

If more detail is wanted (which actual flags are active), another option is to put them in the hover of the environment itself (the same way AB Tests and Preferences does now). I think I'm just hesitant to add a new place where we're adding non-production UI to the screen (what would this look like if we all starting using it, and the list grew to five or more?..).

mreishus committed:
Feature Helper at bottom: Renders in dev only

mreishus committed:
Add enabledFeatures() to config api

Used --no-verify to bypass existing lint violations, unrelated to these
changes:

./wp-calypso/client/lib/create-config/index.js
  41:3  error  Unexpected console statement                             no-console
  61:0  error  Expected @param names to be "data". Got "feature, data"  jsdoc/check-param-names
  64:0  error  Invalid JSDoc tag name "api"                             jsdoc/check-tag-names

mreishus committed:
Feature helper displays both Current XL Projects and Enabled Features

mreishus committed:
Revert "Add Development only flag-notifier"

This reverts commit 70f683d48e119c4b9d4ae6356af7a4bf774b0b56.

mreishus committed:
Merge branch 'master' into add/flag-notifier

mreishus commented:
* Knowing which environment you're on doesn't necessarily tell you if a flag is on. 
  * The (environment x flag) definitions are stored on disk and could change at any time. 
  * URL Overrides could also modify the definitions.
    * Looking at the current URL doesn't necessarily tell you if a URL override is active.  For example, visit `calypso.localhost:3000/home/mysite.wordpress.com?flags=nav-unification`, then click on Posts -> All Posts.  Now the URL reads `http://calypso.localhost:3000/posts/testsite5mmr.wordpress.com`, which doesn't include the flag, but the flag is still active.
  * [Experiment Overrides](https://github.com/Automattic/wp-calypso/pull/47527#) are upcoming feature which also modify the definitions, possibly based on dynamic criteria.

Overall, as we rely more on features for things like gradual rollout, I'd like to have a way to directly check if a feature is on or off without manually computing these things.

Good point about the non-production UIs, I've gone ahead and moved it to that hover bar:
![2020-11-17_17-18](https://user-images.githubusercontent.com/937354/99463154-be9fa280-28fa-11eb-81d0-99c8e0b8cae0.png)

mreishus committed:
Remove unneeded redux provider from featureHelper

mreishus committed:
Feature Helper: Remove unused styles

kwight commented:
> I've gone ahead and moved it to that hover bar

Oooh, I like that more – works for me 👍

mreishus committed:
Merge branch 'master' into add/flag-notifier

mreishus committed:
Fix bug when hiding features helper on non-dev environments

mreishus committed:
spaces -> tabs in features-helper/style.scss

mreishus committed:
Merge branch 'master' into add/flag-notifier

mreishus committed:
Merge branch 'trunk' into add/flag-notifier
